File: projectClient/src/main/java/fpt/aptech/client/models/UserDetails.java
/*
 * Click nbfs://nbhost/SystemFileSystem/Templates/Licenses/license-default.txt to change this license
 * Click nbfs://nbhost/SystemFileSystem/Templates/Classes/Class.java to edit this template
 */
package fpt.aptech.client.models;


import java.util.Date;

/**
 *
 * @author Manh_Chien
 */
public class UserDetails {

   
    public int userdetail_id;
    public Users users;
    public String first_name;
    public String last_name;
    public String email;
    public String phone_number;
    public String address;
    public String note;
    public Date created_dt;

    public UserDetails() {
    }

    public UserDetails(int userdetail_id, Users users, String first_name, String last_name, String email, String phone_number, String address, String note, Date created_dt) {
        this.userdetail_id = userdetail_id;
        this.users = users;
        this.first_name = first_name;
        this.last_name = last_name;
        this.email = email;
        this.phone_number = phone_number;
        this.address = address;
        this.note = note;
        this.created_dt = created_dt;
    }

    public UserDetails( Users users, String first_name, String last_name, String email, String phone_number, String address, String note, Date created_dt) {
        this.users = users;
        this.first_name = first_name;
        this.last_name = last_name;
        this.email = email;
        this.phone_number = phone_number;
        this.address = address;
        this.note = note;
        this.created_dt = created_dt;
    }
    
  

    public int getUserdetail_id() {
        return userdetail_id;
    }

    public void setUserdetail_id(int userdetail_id) {
        this.userdetail_id = userdetail_id;
    }

    public Users getUsers() {
        return users;
    }

    public void setUsers(Users users) {
        this.users = users;
    }

    public String getFirst_name() {
        return first_name;
    }

    public void setFirst_name(String first_name) {
        this.first_name = first_name;
    }

    public String getLast_name() {
        return last_name;
    }

    public void setLast_name(String last_name) {
        this.last_name = last_name;
    }

    public String getEmail() {
        return email;
    }

    public void setEmail(String email) {
        this.email = email;
    }

    public String getPhone_number() {
        return phone_number;
    }

    public void setPhone_number(String phone_number) {
        this.phone_number = phone_number;
    }

    public String getAddress() {
        return address;
    }

    public void setAddress(String address) {
        this.address = address;
    }

    public String getNote() {
        return note;
    }

    public void setNote(String note) {
        this.note = note;
    }

    public Date getCreated_dt() {
        return created_dt;
    }

    public void setCreated_dt(Date created_dt) {
        this.created_dt = created_dt;
    }
    
    

}

File: project4flutter/lib/model/UserDetails.dart
import 'Users.dart';

class UserDetails {
  UserDetails({
    int? userdetailId,
    Users? users,
    String? firstName,
    String? lastName,
    String? email,
    String? phoneNumber,
    String? address,
    String? note,
    DateTime? createdDt,
  })  : _userdetailId = userdetailId,
        _users = users,
        _firstName = firstName,
        _lastName = lastName,
        _email = email,
        _phoneNumber = phoneNumber,
        _address = address,
        _note = note,
        _createdDt = createdDt ?? DateTime.now();

  UserDetails.fromJson(dynamic json)
      : _userdetailId = json['userdetail_id'],
        _users = json['users'] != null ? Users.fromJson(json['users']) : null,
        _firstName = json['first_name'],
        _lastName = json['last_name'],
        _email = json['email'],
        _phoneNumber = json['phone_number'],
        _address = json['address'],
        _note = json['note'],
        _createdDt = _parseDate(json['created_dt']);

  final int? _userdetailId;
  final Users? _users;
  final String? _firstName;
  final String? _lastName;
  final String? _email;
  final String? _phoneNumber;
  final String? _address;
  final String? _note;
  final DateTime? _createdDt;

  int get userdetailId => _userdetailId ?? 0;
  Users? get users => _users;
  String get firstName => _firstName ?? '';
  String get lastName => _lastName ?? '';
  String get email => _email ?? '';
  String get phoneNumber => _phoneNumber ?? '';
  String get address => _address ?? '';
  String get note => _note ?? '';
  String get createdDt => _formatDate(_createdDt) ?? '';

  Map<String, dynamic> toJson() {
    final map = <String, dynamic>{};
    map['userdetail_id'] = _userdetailId;
    map['users'] = _users?.toJson();
    map['first_name'] = _firstName;
    map['last_name'] = _lastName;
    map['email'] = _email;
    map['phone_number'] = _phoneNumber;
    map['address'] = _address;
    map['note'] = _note;
    map['created_dt'] = _formatDate(_createdDt);
    return map;
  }

  static DateTime? _parseDate(String? dateStr) {
    if (dateStr == null || dateStr.isEmpty) return null;
    try {
      return DateTime.parse(dateStr);
    } catch (e) {
      print('Error parsing date: $e');
      return null;
    }
  }

  static String? _formatDate(DateTime? date) {
    if (date == null) return null;
    return date.toIso8601String();
  }
}
